app/controllers/users/backup_code_setup_controller.rb

```diff
@@ -112,6 +112,10 @@ def user_opted_remember_device_cookie
     end
 
     def mark_user_as_fully_authenticated
+      if user_session[:auth_method].blank?
```

Comment:
Why is this conditional? (Same note for PIV)

Reply:
Ah oops, forgot this was in there. I was experimenting with whether we always want to set it

e.g. if I already have authenticated with my PIV or something, do we want to "downgrade" my auth_method to backup codes. If I then attempt to authenticate to a phishing-resistant or PIV-required SP, I'll be asked to verify again. This is already how it works now for other methods, so I don't think it's adding any _new_ problems, but it's something we could sorta address for backup codes since they'll typically be configured last on account setup?

Reply:
Hm, maybe we need a way to track multiple authentication methods? 

One concern I might have with this is that it can refresh `authn_at` for a different `auth_method`. Is that a problem?

Reply:
That's a good point. I don't think it would necessarily cause any problems today, but it's better to be consistent while we have these updates scattered around for now.

`authn_at` is only used for re-authentication checks currently (which is entirely internal functionality at the moment). 
We may want to keep track of it more comprehensively as an array or something with whether it was confirmation/setup or an authentication, timestamp, etc. It could help us protect against edge cases where a weaker method overwrites a stronger one that would meet phishing-resistant or HSPD12 and prevent a user from having to authenticate again. I'm hazarding a guess that it's not particularly frequent, but worth addressing in another PR either way I think.

I've removed the `if user_session[:auth_method].blank?` check to keep the behavior consistent across controllers.